charts/vector/templates/haproxy/configmap.yaml

```diff
@@ -7,7 +7,7 @@ metadata:
     {{- include "haproxy.labels" . | nindent 4 }}
 data:
   haproxy.cfg: |
-  {{- if .Values.customConfig }}
+  {{- if .Values.haproxy.customConfig }}
```

Comment:
Looking at this I think we actually need a bit more of a robust checking. If Vector is using a non-default config, the default config provided for HAProxy isn't guaranteed to route correctly - and the reverse is also true.

Perhaps checking if either are custom and raising if that's the case would be helpful - what do you think?